src/main/java/chatbot/Parser.java

```diff
@@ -0,0 +1,142 @@
+package chatbot;
+
+import chatbot.command.*;
+
+/**
+ * Class which deals with making sense of the user command.
+ */
+public class Parser {
+
+    /**
+     * Enumeration representing different types of tasks and commands.
+     */
+    public enum TaskType {
+        TODO, DEADLINE, EVENT, MARK, UNMARK, UNKNOWN, BYE;
+    }
+
+    /**
+     * method to determine types of the task.
+     *
+     * @param userInput user command
+     * @return Task type of the command
+     */
+    public static TaskType determineTaskType(String userInput) {
+        if (userInput.startsWith("todo")) return TaskType.TODO;
+        if (userInput.startsWith("deadline")) return TaskType.DEADLINE;
+        if (userInput.startsWith("event")) return TaskType.EVENT;
+        if (userInput.startsWith("mark")) return TaskType.MARK;
+        if (userInput.startsWith("unmark")) return TaskType.UNMARK;
+        if ("bye".equalsIgnoreCase(userInput)) return TaskType.BYE;
+        return TaskType.UNKNOWN;
+    }
+
+    /**
+     * Parse the user command which its task type is todo.
+     *
+     * @param userInput user command
+     * @return array of string after parsing the command
+     * @throws ChatbotException if the description is empty
+     */
+    public static String[] parseTodo(String userInput) throws ChatbotException {
+        if (userInput.length() <= 4) {
+            throw new ChatbotException("☹ OOPS!!! The description of a todo cannot be empty.");
+        }
+        String taskDescription = userInput.substring(5);
+        return new String[]{taskDescription};
+    }
+
+    /**
+     * Parse the user command which its task type is DEADLINE.
+     *
+     * @param userInput user command
+     * @return array of string after parsing the command
+     * @throws ChatbotException if the description or date is empty
+     */
+    public static String[] parseDeadline(String userInput) throws ChatbotException {
```

Comment:
I like how you have accounted for specific types of errors within each task parser function!